src/app/reporting/Engine.h

```diff
@@ -118,6 +118,13 @@ class Engine
                                    const ConcreteReadAttributePath & aClusterInfo,
                                    AttributeValueEncoder::AttributeEncodeState * apEncoderState);
 
+    // if version match, it means don't send, if version mismatch, it means send.
+    // if client sends the same path with multiple data versions, client will get the data back per the spec, because at least one
+    // of those will fail to match It should return false if either nothing in the list matches the given endpoint+cluster in the
+    // path or there is an entry in the list that matches the endpoint+cluster in the path but does not match the current data
+    // version of that cluster. .
```

Comment:
```suggestion
    // If version match, it means don't send, if version mismatch, it means send.
    // If client sends the same path with multiple data versions, client will get the data back per the spec, because at least one
    // of those will fail to match.  This function should return false if either nothing in the list matches the given endpoint+cluster in the
    // path or there is an entry in the list that matches the endpoint+cluster in the path but does not match the current data
    // version of that cluster. 
```

Reply:
Done, thanks